### DISCUSSION TITLE
Popup frame sizing cutting off bottom rows

### DISCUSSION THREAD

**Discussion by shipmints** [minad/corfu#474] [Q&A] [answered]
Thank you for this great package and all your other great packages.

I know similar issues have been raised in several other incarnations and I have looked at them. I'm not sure if the use cases were as simple as this one which seems likely easily reproducible by anyone.

This case simply sets the corfu-default face height to a smaller scale than 1.0 If I set the height scale to 1.0, it's better but still cuts off just a little bit.

I took a quick look at the code to see how it determines the pop-up frame size and it looks like it takes line height into account but not face height? Are those equivalized inside Emacs? I'm not an expert with the internals as much as you are (you are clearly an expert).

```
(use-package corfu
...
  :custom-face
  (corfu-default ((t (:height 0.85))))
...
```
This is Emacs 29.3 on MacOS using the latest [m]elpa corfu-20240704.1510. The theme is modus-vivendi-deuteranopia, but I don't think that has any bearing.

<img width="562" alt="image" src="https://github.com/minad/corfu/assets/30732365/92c3c973-8f2b-423a-8a46-3068fd83c52b">

**Comment by minad** [comment, marked as answer]
From what I can tell the problem here might be the icons. This exact problem has been described before and can be fixed by correcting the icon height.

**Reply by shipmints** [reply]
I do already adjust icon height (or the screenshot would have looked very skewed) so that's not the solution. Unless I should scale the icons down even more than the corfu font?

```elisp
  (setq kind-icon-default-style `(:padding -1 :stroke 0 :margin 0 :radius 0 :scale 1.0 :height 0.85)) ; make sure icons fit with scaled text
```

**Reply by shipmints** [reply]
Just tested that and it does work. I changed the height scale of the fonts to 0.60 vs. the corfu height of 0.85.

Now all kind-icons are scaled down to tiny icons everywhere they are used. So I'll have to work on some method to scale down for corfu pop-ups only.

Thank you, @minad

**Reply by jdtsmith** [reply]
Note that there is a bug in svg-lib on some hi-DPI systems that leads to "double height" SVG icons (rougier/svg-lib#46).
